### PR TITLE
fix: 1295 udaf default value for `min_where` and `max_where`

### LINKED ISSUE #1295: in the aggregate statistics operation after filtering, when the filter conditions are not matched, the "max_where" and "min_where" display results are abnormal
Test environment: cluster, Mac ,online
The reproduction process is as follows:
      1.  create table t1 (id bigint, uid string, mid string, cardno string, trans_time timestamp, trans_amt double, trans_type string,        province string ,city string, label bool);
image
     2.  
image
      <When the data of trans_type = "POS" cannot be matched, there is a problem with the result display.>

### PR DESCRIPTION
**Pull request by aceforeverd**
close #1295 

rules: 
1. if no cols matched, return `NULL`
2. if all cols matched is `NULL`, return `NULL`

### CODE REVIEW

**Comment by aceforeverd** [comment]
[bugfix-liguo](https://trello.com/c/Ut8UamBY/37-bugfix-liguo)

File: hybridse/src/udf/udaf_test.cc [hunk 1]
```diff
@@ -52,6 +52,57 @@ void CheckUdfFail(const std::string &name, T expect, Args... args) {
...
+void CheckUdafOneParam(const std::string &fn, Ret expect, std::initializer_list<Arg> cols) {
+    CheckUdf(fn, expect, MakeList(cols));
+}
+
+TEST_F(UdafTest, MaxTest) {
+    CheckUdafOneParam<Nullable<int32_t>>("max", nullptr, {});
+    CheckUdafOneParam<Nullable<int32_t>, Nullable<int32_t>>("max", nullptr, {nullptr});
+    CheckUdafOneParam<Nullable<int32_t>, Nullable<int32_t>>("max", nullptr, {nullptr, nullptr});
+    CheckUdafOneParam<Nullable<int64_t>, Nullable<int64_t>>("max", 5, {nullptr, 5, 1});
+    CheckUdafOneParam<Nullable<int64_t>, Nullable<int64_t>>("max", 5, {nullptr, 5, 1});
```
Comment (zhanghaohit) [inline hybridse/src/udf/udaf_test.cc:66]: L65 and L66 are the same?

File: hybridse/src/udf/udaf_test.cc [hunk 2]
```diff
@@ -52,6 +52,57 @@ void CheckUdfFail(const std::string &name, T expect, Args... args) {
...
+    CheckUdafOneParam<Nullable<StringRef>, Nullable<StringRef>>("max", StringRef("abc"),
+                                                                {nullptr, StringRef("abc"), StringRef("aaa")});
+}
+
+TEST_F(UdafTest, MinTest) {
+    CheckUdafOneParam<Nullable<int32_t>>("min", nullptr, {});
+    CheckUdafOneParam<Nullable<int32_t>, Nullable<int32_t>>("min", nullptr, {nullptr});
+    CheckUdafOneParam<Nullable<int32_t>, Nullable<int32_t>>("min", nullptr, {nullptr, nullptr});
+    CheckUdafOneParam<Nullable<int64_t>, Nullable<int64_t>>("min", 1, {nullptr, 5, 1});
+    CheckUdafOneParam<Nullable<int64_t>, Nullable<int64_t>>("min", 1, {nullptr, 5, 1});
```
Comment (zhanghaohit) [inline hybridse/src/udf/udaf_test.cc:80]: also here

File: hybridse/src/udf/udaf_test.cc (hunk 1 above)
Comment (aceforeverd) [reply hybridse/src/udf/udaf_test.cc:66]: fixed

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**